src/firestore/collection/collection.ts

```diff
@@ -89,7 +90,7 @@ export class AngularFirestoreCollection<T> {
    */
   snapshotChanges(events?: firebase.firestore.DocumentChangeType[]): Observable<DocumentChangeAction[]> {
```

Comment:
We need the `.share()` operator further down in `fromRef.ts`. That way each observable method benefits from a "shared" source.

```ts
export function fromRef<R>(ref: firebase.firestore.DocumentReference | firebase.firestore.Query) {
  return _fromRef<typeof ref, R>(ref).share();
}
```

Reply:
SGTM, but as a optimization.
As mentioned, we need the share on `sortedChanges` (which I'll bump up to the definition) to make the side effects (`combineChanges`) occur only the once.